warehouse/integrations/snowflake/snowflake.go

```diff
@@ -1062,50 +1047,52 @@ func (sf *Snowflake) LoadUserTables(ctx context.Context) map[string]error {
 }
 
 func (sf *Snowflake) connect(ctx context.Context, opts optionalCreds) (*sqlmw.DB, error) {
-	cred := sf.getConnectionCredentials(opts)
-	urlConfig := snowflake.Config{
-		Account:     cred.account,
-		User:        cred.user,
-		Role:        cred.role,
-		Database:    cred.database,
-		Schema:      cred.schemaName,
-		Warehouse:   cred.warehouse,
-		Application: application,
-	}
-	if cred.timeout > 0 {
-		urlConfig.LoginTimeout = cred.timeout
-	}
-	if cred.useKeyPairAuth {
-		rsaPrivateKey, err := ParsePrivateKey(cred.privateKey, cred.privateKeyPassphrase)
-		if err != nil {
-			return nil, fmt.Errorf("parsing private key: %w", err)
-		}
-
-		urlConfig.PrivateKey = rsaPrivateKey
-		urlConfig.Authenticator = snowflake.AuthTypeJwt
-	} else {
-		urlConfig.Password = cred.password
-		urlConfig.Authenticator = snowflake.AuthTypeSnowflake
-	}
+	var (
+		account              = whutils.GetConfigValue(configKeyAccount, sf.Warehouse)
```

Comment:
Not introduced here, but I noticed how GetConfigValue works.

1. Do we really need: `	configKey := fmt.Sprintf("Warehouse.pipeline.%s.%s.%s", warehouse.Source.ID, warehouse.Destination.ID, key)`
2. Can we replace it with a non global config method?

Reply:
Regarding 1, this was added as providing pipeline level sync frequency override in [here](https://github.com/rudderlabs/rudder-server/pull/3094).

Regarding 2, Yeah we can do that but that would require passing *`config.Config` as a parameter, something like this:
```GO
func GetConfigValue(conf *config.Config, key string, warehouse model.Warehouse) (val string) {
	configKey := fmt.Sprintf("Warehouse.pipeline.%s.%s.%s", warehouse.Source.ID, warehouse.Destination.ID, key)
	if conf.IsSet(configKey) {
		return conf.GetString(configKey, "")
	}
	destConfig := warehouse.Destination.Config
	if destConfig[key] != nil {
		val, _ = destConfig[key].(string)
	}
	return val
}
```

Reply:
Regarding 2, I can take it as part of a separate task.

Reply:
https://linear.app/rudderstack/issue/PIPE-1365/handle-review-comment-for-global-config